### PR TITLE
optimized sram usage (about 40% of what it was)

### PR DESCRIPTION
**Pull request by cambierr**
I don't know if it makes sens to merge my changes in the main lib but feel free to accept this PR if you think it does.

What I actually changed is how strings are stored (mainly for debug messages, bug also for chip commands) to lower the library SRAM consumption to less than half of it's current memory usage.

It's not perfect code (it's possible to do it butch better by changing the lib functions) but hey, this is a quick and working, fix!

### CODE REVIEW

**Comment by FokkeZB** [comment]
**FR** complete

Tested Get-Device-Info and hello-world-ABP and they still work as expected

@johanstokking could you code review and merge? I'll tag this as v0.8 and apply for it to be included as built-in library then.

**Comment by johanstokking** [comment]
Thanks @cambierr. Looks good. I'll test with OTAA once I'm close to a gateway again. Expect next week.

**Comment by cambierr** [comment]
You're welcome :-) I just wrote that to fix issues we had yesterday during a meetup.
Btw, is downlink support gone ?

**Comment by johanstokking** [comment]
We went back in time, will raise issue for this, thanks

**Comment by johanstokking** [comment]
@cambierr this is fixed now (see #2). Can you rebase?

**Comment by FokkeZB** [comment]
Sorry for that @cambierr, I've fixed #2, doing some final test but go ahead and rebase

**Comment by FokkeZB** [comment]
@cambierr sorry for the mess but it seems you rebased just before I did some other changes (renaming `ttu` to `ttn`) that are missing in your PR. Could you check that?

**Comment by cambierr** [comment]
Done!

**Comment by cambierr** [comment]
what ttu/ttn ? I can't see any ?

File: examples/DeviceInfo/DeviceInfo.ino [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #define debugSerial Serial
 #define loraSerial Serial1
 
-TheThingsNetwork ttn;
```
Comment (FokkeZB) [inline examples/DeviceInfo/DeviceInfo.ino]: This (and the other places) should be left as `ttn`, not `ttu` as it is currently in e.g. https://github.com/TheThingsNetwork/arduino-library/blob/master/examples/DeviceInfo/DeviceInfo.ino#L6

**Comment by FokkeZB** [comment]
@cambierr in the end, the only file you wanted to change is https://github.com/TheThingsNetwork/arduino-library/blob/master/src/TheThingsNetwork.cpp right?

**Comment by cambierr** [comment]
So, this time my fork is synced with your last updates !

**Comment by FokkeZB** [comment]
Thanks for your patience! Once #3 is solved @johanstokking can double-check and merge

**Comment by FokkeZB** [comment]
- Fixed #3 
- Verified PR with DeviceInfo, SendOTAA and Downlink

@johanstokking yours to merge

**Comment by johanstokking** [comment]
Thanks @cambierr and @FokkeZB. Highly appreciated.

**Pull request merged**